# Problem Statement
### Problem background
Suppose that you are working as a data analyst at Airbnb. For the past few months, Airbnb has seen a major decline in revenue. Now that the restrictions have started lifting and people have started to travel more, Airbnb wants to make sure that it is fully prepared for this change.<br><br>

<h6>The different leaders at Airbnb want to understand some important insights based on various attributes in the dataset so as to increase the revenue such as -<h6>

1. Which type of hosts to acquire more and where?
2. The categorisation of customers based on their preferences.
    
    * What are the neighbourhoods they need to target?
    * What is the pricing ranges preferred by customers?
    * The various kinds of properties that exist w.r.t. customer preferences.
    * Adjustments in the existing properties to make it more customer-oriented.
3. What are the most popular localities and properties in New York currently?
4. How to get unpopular properties more traction? and so on...<br><br><br>
<b>Note:</b> These points are just to give you an initial understanding of how to proceed with the analysis and the kind of questions you need to ask. In addition to this, you would need to further understand the data (attached below in this page) and analyse important insights that you feel are required for the audiences mentioned for each of the two presentations.

 

### End Objective
To prepare for the next best steps that Airbnb needs to take as a business, you have been asked to analyse a dataset consisting of various Airbnb listings in New York. Based on this analysis, you need to give two presentations to the following groups.<br>

* Presentation - I
    * Data Analysis Managers: These people manage the data analysts directly for processes and their technical expertise is basic.
    * Lead Data Analyst: The lead data analyst looks after the entire team of data and business analysts and is technically sound.
* Presentation - II
    * Head of Acquisitions and Operations, NYC: This head looks after all the property and host acquisitions and operations. Acquisition of the best properties, price negotiation, and negotiating the services the properties offer falls under the purview of this role.
    * Head of User Experience, NYC: The head of user experience looks after the customer preferences and also handles the properties listed on the website and the Airbnb app. Basically, the head of user experience tries to optimise the order of property listing in certain neighbourhoods and cities in order to get every property the optimal amount of traction.

# Some tips before starting the case study
<h6>Please go through all of the points mentioned below carefully.</h6>

1. The case study has been kept deliberately vague with little background information. So don't get worried and make sure you perform a thorough analysis of the data and extract the most meaningful insights.<br><br>
2. Many a time you do get such loosely presented business problems where you don't have much business background. In such cases, it is important to stick to the basics and perform all data cleaning, analysis and visualisation processes structurally.<br><br>
3. This assignment is different from the ones you have solved before. Make sure that you treat this case study as a storytelling exercise and not an analysis/visualisation one. This will help you be better prepared for the presentations.<br><br>
4. Although this is a storytelling exercise meaning that the tasks performed on the data are not important for the end objective, there is a methodology document that you need to submit. This document should ideally contain the entire methodology, technical details, and assumptions wherever applicable.<br><br> 
5. This document is also necessary to verify all the steps you have performed to arrive at the final presentations and hence without it, your presentations will not be graded.
The data cleaning, analysis, and visualisation part can be done on any tool(s) of your choice. But again, make sure that you are documenting all the steps, assumptions, and major codes/plots.<br><br>
6. Once you are done with the analysis and visualisations, there will be many insights at your hand. Make sure that you map the right visuals and takeaways with the right audience since some of these insights might be relevant to one group but not to the other group.<br><br>
7. Recall all the elements mentioned by Utkalika and always try to follow the rules to tell a good data story while creating the presentation.<br><br>
8. Most importantly, have fun solving this case study!

In [1]:
import numpy as np, pandas as pd, matplotlib.pyplot as ply, seaborn as sns, plotly.graph_objects as go

In [2]:
import os
os.chdir('C:\\Users\\91703\\Downloads')


In [3]:
# setting the dataframe truncation to none
pd.set_option('max_columns',None,'max_rows',None,'display.width',None,'display.max_colwidth',None,'display.max_rows',None)   

In [4]:
airbnb = pd.read_csv('AB_NYC_2019.csv')

In [5]:
airbnb.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  19-10-2018   
1  Entire home/apt    225               1                 45  21-05-2019   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  05-07-2019   
4  Entire home/apt     80              10                  9  19-11-2018   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [6]:
airbnb.shape

(48895, 16)

In [7]:
airbnb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

<b><em><font color='salmon'>As last_review column is date we are parsing it as date and reading the df again</font></em></b>

In [8]:
airbnb = pd.read_csv('AB_NYC_2019.csv', parse_dates=[12])

In [9]:
airbnb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype         
---  ------                          --------------  -----         
 0   id                              48895 non-null  int64         
 1   name                            48879 non-null  object        
 2   host_id                         48895 non-null  int64         
 3   host_name                       48874 non-null  object        
 4   neighbourhood_group             48895 non-null  object        
 5   neighbourhood                   48895 non-null  object        
 6   latitude                        48895 non-null  float64       
 7   longitude                       48895 non-null  float64       
 8   room_type                       48895 non-null  object        
 9   price                           48895 non-null  int64         
 10  minimum_nights                  48895 non-null  int64         
 11  nu

In [10]:
# checking statistical summary
airbnb.describe([0.01,0.25,0.5,0.75,0.9,0.95])  

id       host_id      latitude     longitude         price  \
count  4.889500e+04  4.889500e+04  48895.000000  48895.000000  48895.000000   
mean   1.901714e+07  6.762001e+07     40.728949    -73.952170    152.720687   
std    1.098311e+07  7.861097e+07      0.054530      0.046157    240.154170   
min    2.539000e+03  2.438000e+03     40.499790    -74.244420      0.000000   
1%     1.717151e+05  1.006922e+05     40.596687    -74.026774     30.000000   
25%    9.471945e+06  7.822033e+06     40.690100    -73.983070     69.000000   
50%    1.967728e+07  3.079382e+07     40.723070    -73.955680    106.000000   
75%    2.915218e+07  1.074344e+08     40.763115    -73.936275    175.000000   
90%    3.399094e+07  2.128114e+08     40.804890    -73.907810    269.000000   
95%    3.525910e+07  2.417646e+08     40.825643    -73.865771    355.000000   
max    3.648724e+07  2.743213e+08     40.913060    -73.712990  10000.000000   

       minimum_nights  number_of_reviews  reviews_per_month  \
count    48895.000000       48895.000000       38843.000000   
mean         7.029962          23.274466           1.373221   
std         20.510550          44.550582           1.680442   
min          1.000000           0.000000           0.010000   
1%           1.000000           0.000000           0.020000   
25%          1.000000           1.000000           0.190000   
50%          3.000000           5.000000           0.720000   
75%          5.000000          24.000000           2.020000   
90%         28.000000          70.000000           3.630000   
95%         30.000000         114.000000           4.640000   
max       1250.000000         629.000000          58.500000   

       calculated_host_listings_count  availability_365  
count                    48895.000000      48895.000000  
mean                         7.143982        112.781327  
std                         32.952519        131.622289  
min                          1.000000          0.000000  
1%                           1.000000          0.000000  
25%                          1.000000          0.000000  
50%                          1.000000         45.000000  
75%                          2.000000        227.000000  
90%                          5.000000        337.000000  
95%                         15.000000        359.000000  
max                        327.000000        365.000000

In [11]:
airbnb.isnull().sum()/len(airbnb)*100

id                                 0.000000
name                               0.032723
host_id                            0.000000
host_name                          0.042949
neighbourhood_group                0.000000
neighbourhood                      0.000000
latitude                           0.000000
longitude                          0.000000
room_type                          0.000000
price                              0.000000
minimum_nights                     0.000000
number_of_reviews                  0.000000
last_review                       20.558339
reviews_per_month                 20.558339
calculated_host_listings_count     0.000000
availability_365                   0.000000
dtype: float64

###### name

In [12]:
airbnb[airbnb.name.isnull()]

id name   host_id host_name neighbourhood_group  \
2854    1615764  NaN   6676776     Peter           Manhattan   
3703    2232600  NaN  11395220      Anna           Manhattan   
5775    4209595  NaN  20700823     Jesse           Manhattan   
5975    4370230  NaN  22686810   Michaël           Manhattan   
6269    4581788  NaN  21600904     Lucie            Brooklyn   
6567    4756856  NaN   1832442  Carolina            Brooklyn   
6605    4774658  NaN  24625694      Josh           Manhattan   
8841    6782407  NaN  31147528  Huei-Yin            Brooklyn   
11963   9325951  NaN  33377685  Jonathan           Manhattan   
12824   9787590  NaN  50448556    Miguel           Manhattan   
13059   9885866  NaN  37306329  Juliette           Manhattan   
13401  10052289  NaN  49522403   Vanessa            Brooklyn   
15819  12797684  NaN  69715276       Yan           Manhattan   
16071  12988898  NaN  71552588    Andrea               Bronx   
18047  14135050  NaN  85288337      Jeff            Brooklyn   
28889  22275821  NaN  49662398  Kathleen            Brooklyn   

            neighbourhood  latitude  longitude        room_type  price  \
2854    Battery Park City  40.71239  -74.01620  Entire home/apt    400   
3703         East Village  40.73215  -73.98821  Entire home/apt    200   
5775    Greenwich Village  40.73473  -73.99244  Entire home/apt    225   
5975               Nolita  40.72046  -73.99550  Entire home/apt    215   
6269         Williamsburg  40.71370  -73.94378     Private room    150   
6567             Bushwick  40.70046  -73.92825     Private room     70   
6605   Washington Heights  40.85198  -73.93108     Private room     40   
8841         Williamsburg  40.71354  -73.93882     Private room     45   
11963      Hell's Kitchen  40.76436  -73.98573  Entire home/apt    190   
12824              Harlem  40.80316  -73.95189  Entire home/apt    300   
13059           Chinatown  40.71632  -73.99328     Private room     67   
13401         Brownsville  40.66409  -73.92314     Private room     50   
15819     Upper West Side  40.79843  -73.96404     Private room    100   
16071             Fordham  40.86032  -73.88493      Shared room    130   
18047  Bedford-Stuyvesant  40.69421  -73.93234     Private room     70   
28889            Bushwick  40.69546  -73.92741  Entire home/apt    110   

       minimum_nights  number_of_reviews last_review  reviews_per_month  \
2854             1000                  0         NaT                NaN   
3703                1                 28  2015-08-06               0.45   
5775                1                  1  2015-01-01               0.02   
5975                7                  5  2016-02-01               0.09   
6269                1                  0         NaT                NaN   
6567                1                  0         NaT                NaN   
6605                1                  0         NaT                NaN   
8841                1                  0         NaT                NaN   
11963               4                  1  2016-05-01               0.02   
12824               5                  0         NaT                NaN   
13059               4                  0         NaT                NaN   
13401               3                  3  2016-08-18               0.07   
15819               1                  0         NaT                NaN   
16071               1                  0         NaT                NaN   
18047               3                  0         NaT                NaN   
28889               4                  5  2018-08-13               0.27   

       calculated_host_listings_count  availability_365  
2854                                1               362  
3703                                1               341  
5775                                1                 0  
5975                                1                 0  
6269                                1                 0  
6567                                1    

###### host_name

In [13]:
airbnb[airbnb.host_name.isnull()]

id                                                name  \
360      100184                                           Bienvenue   
2700    1449546                             Cozy Studio in Flatbush   
5745    4183989     SPRING in the City!! Zen-Style Tranquil Bedroom   
6075    4446862                  Charming Room in Prospect Heights!   
6582    4763327                 Luxurious, best location, spa inc'l   
8163    6292866                           Modern Quiet Gem Near All   
8257    6360224                     Sunny, Private room in Bushwick   
8852    6786181                        R&S Modern Spacious Hideaway   
9138    6992973                     1 Bedroom in Prime Williamsburg   
9817    7556587                                Sunny Room in Harlem   
14040  10709846                  Sunny, spacious room in Greenpoint   
14631  11553543                                   Cozy Room Astoria   
15174  12113879      Sunny, Large West Village 1 BR Near Everything   
19565  15648096               Spacious 2 bedroom close to Manhattan   
27777  21787094          Spacious and Bright Midtown East Apartment   
27962  21867710   Amazing room in the centre on Manhatten! Welcome!   
28042  21901156  Best Deal! Lovely place in Manhattan! Time Square!   
28274  21990808                  ***SLEEPING IN THE GOLDEN PLACE***   
30570  23660605     Private Bedroom w/ Balcony view of Central Park   
32193  25180572                      Beautiful Cozy Studio Apt. UWS   
38992  30468335             LARGE Prospect Lefferts Gardens Bedroom   

         host_id host_name neighbourhood_group     neighbourhood  latitude  \
360       526653       NaN              Queens    Queens Village  40.72413   
2700     7779204       NaN            Brooklyn          Flatbush  40.64965   
5745      919218       NaN           Manhattan            Harlem  40.80606   
6075    23077718       NaN            Brooklyn     Crown Heights  40.67512   
6582    24576978       NaN            Brooklyn        Greenpoint  40.72035   
8163    32722063       NaN            Brooklyn     East Flatbush  40.65263   
8257    33134899       NaN            Brooklyn          Bushwick  40.70146   
8852    32722063       NaN            Brooklyn     East Flatbush  40.64345   
9138     5162530       NaN            Brooklyn      Williamsburg  40.71838   
9817    39608626       NaN           Manhattan            Harlem  40.82929   
14040    7822683       NaN            Brooklyn        Greenpoint  40.73539   
14631   26138712       NaN              Queens  Ditmars Steinway  40.77587   
15174    5300585       NaN           Manhattan           Chelsea  40.73949   
19565  100971588       NaN               Bronx        Highbridge  40.83844   
27777     415290       NaN           Manhattan   Upper East Side  40.76436   
27962  159156636       NaN           Manhattan    Hell's Kitchen  40.75668   
28042  159156636       NaN           Manhattan    Hell's Kitchen  40.75656   
28274  159156636       NaN           Manhattan    Hell's Kitchen  40.75835   
30570  177146433       NaN           Manhattan       East Harlem  40.79766   
32193  119609345       NaN           Manhattan   Upper West Side  40.77701   
38992  228750026       NaN            Brooklyn          Flatbush  40.65152   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
360    -73.76133     Private room     50               1                 43   
2700   -73.96154  Entire home/apt    100              30                 49   
5745   -73.95061     Private room     86               3                 34   
6075   -73.96146     Private room     50               1                  0   
6582   -73.95355  Entire home/apt    195               1                  1   
8163   -73.93215  Entire home/apt     85               2                182   
8257   -73.92792     Private room     37               1                  1   
8852   -73.93643  Entire home/apt    100               2                157   
9138   -73.95630  Enti

In [14]:
1 - airbnb[airbnb.host_name.isnull() | airbnb.name.isnull()].shape[0]/len(airbnb)

0.9992432764086308

<b><em><font color='salmon'>We can drop all rows with name and host_name as nan; we still retain 99.9% data</font></em></b>

###  dropping those rows

In [41]:
airbnb = airbnb[-(airbnb.host_name.isnull() | airbnb.name.isnull())]

### last_review & reviews_per_month

In [15]:
airbnb[airbnb.last_review.isnull()].head()

id                                               name  host_id  \
2    3647                THE VILLAGE OF HARLEM....NEW YORK !     4632   
19   7750                  Huge 2 BR Upper East  Cental Park    17985   
26   8700  Magnifique Suite au N de Manhattan - vue Cloitres    26394   
36  11452                        Clean and Quiet in Brooklyn     7355   
38  11943                          Country space in the city    45445   

          host_name neighbourhood_group       neighbourhood  latitude  \
2         Elisabeth           Manhattan              Harlem  40.80902   
19             Sing           Manhattan         East Harlem  40.79685   
26  Claude & Sophie           Manhattan              Inwood  40.86754   
36               Vt            Brooklyn  Bedford-Stuyvesant  40.68876   
38          Harriet            Brooklyn            Flatbush  40.63702   

    longitude        room_type  price  minimum_nights  number_of_reviews  \
2   -73.94190     Private room    150               3                  0   
19  -73.94872  Entire home/apt    190               7                  0   
26  -73.92639     Private room     80               4                  0   
36  -73.94312     Private room     35              60                  0   
38  -73.96327     Private room    150               1                  0   

   last_review  reviews_per_month  calculated_host_listings_count  \
2          NaT                NaN                               1   
19         NaT                NaN                               2   
26         NaT                NaN                               1   
36         NaT                NaN                               1   
38         NaT                NaN                               1   

    availability_365  
2                365  
19               249  
26                 0  
36               365  
38               365

In [16]:
airbnb[airbnb.reviews_per_month.isnull()].head()

id                                               name  host_id  \
2    3647                THE VILLAGE OF HARLEM....NEW YORK !     4632   
19   7750                  Huge 2 BR Upper East  Cental Park    17985   
26   8700  Magnifique Suite au N de Manhattan - vue Cloitres    26394   
36  11452                        Clean and Quiet in Brooklyn     7355   
38  11943                          Country space in the city    45445   

          host_name neighbourhood_group       neighbourhood  latitude  \
2         Elisabeth           Manhattan              Harlem  40.80902   
19             Sing           Manhattan         East Harlem  40.79685   
26  Claude & Sophie           Manhattan              Inwood  40.86754   
36               Vt            Brooklyn  Bedford-Stuyvesant  40.68876   
38          Harriet            Brooklyn            Flatbush  40.63702   

    longitude        room_type  price  minimum_nights  number_of_reviews  \
2   -73.94190     Private room    150               3                  0   
19  -73.94872  Entire home/apt    190               7                  0   
26  -73.92639     Private room     80               4                  0   
36  -73.94312     Private room     35              60                  0   
38  -73.96327     Private room    150               1                  0   

   last_review  reviews_per_month  calculated_host_listings_count  \
2          NaT                NaN                               1   
19         NaT                NaN                               2   
26         NaT                NaN                               1   
36         NaT                NaN                               1   
38         NaT                NaN                               1   

    availability_365  
2                365  
19               249  
26                 0  
36               365  
38               365

In [17]:
airbnb[airbnb.last_review.isnull() & airbnb.reviews_per_month.isnull()].shape

(10052, 16)

In [18]:
airbnb[airbnb.reviews_per_month.isnull()].shape

(10052, 16)

In [19]:
airbnb[airbnb.last_review.isnull()].shape

(10052, 16)

<b><em><font color='salmon'>We can observe that all the rows which have last_review as nan also have reviews_per_month as nan</font></em></b>

In [20]:
airbnb.last_review.describe()

<ipython-input-20-7dfdc1555c78>:1: FutureWarning: Treating datetime data as categorical rather than numeric in `.describe` is deprecated and will be removed in a future version of pandas. Specify `datetime_is_numeric=True` to silence this warning and adopt the future behavior now.
  airbnb.last_review.describe()


count                   38843
unique                   1764
top       2019-06-23 00:00:00
freq                     1413
first     2011-03-28 00:00:00
last      2019-12-06 00:00:00
Name: last_review, dtype: object

### creating a new df where last_review and reviews_per_month are nan

In [24]:
temp = airbnb[airbnb.last_review.isnull() & airbnb.reviews_per_month.isnull()]

In [25]:
temp.shape

(10052, 16)

In [26]:
temp.describe()

id       host_id      latitude     longitude         price  \
count  1.005200e+04  1.005200e+04  10052.000000  10052.000000  10052.000000   
mean   2.257484e+07  8.068437e+07     40.732099    -73.956117    192.919021   
std    1.135563e+07  8.712593e+07      0.052598      0.043796    358.653017   
min    3.647000e+03  4.632000e+03     40.499790    -74.242850      0.000000   
25%    1.208404e+07  1.207519e+07     40.697570    -73.984758     70.000000   
50%    2.337757e+07  3.979538e+07     40.728870    -73.960175    120.000000   
75%    3.402092e+07  1.330001e+08     40.763643    -73.939877    200.000000   
max    3.648724e+07  2.743213e+08     40.911690    -73.716900  10000.000000   

       minimum_nights  number_of_reviews  reviews_per_month  \
count    10052.000000            10052.0                0.0   
mean        11.519797                0.0                NaN   
std         29.207928                0.0                NaN   
min          1.000000                0.0                NaN   
25%          1.000000                0.0                NaN   
50%          3.000000                0.0                NaN   
75%         14.000000                0.0                NaN   
max       1000.000000                0.0                NaN   

       calculated_host_listings_count  availability_365  
count                    10052.000000      10052.000000  
mean                        14.793275        104.660466  
std                         50.363637        139.072127  
min                          1.000000          0.000000  
25%                          1.000000          0.000000  
50%                          1.000000          6.000000  
75%                          2.000000        214.250000  
max                        327.000000        365.000000

<b><em><font color='salmon'>we can see that the number_of_reviews are 0 when last_review & review_per_month are nan;
    This means they never recieved reviews<br><br>
    we can change reviews_per_month to 0 for such cases<br>
    and we can set a dummy date in last_review so we will retain the data while maintaining its quality
    
</font></em></b>

In [27]:
# changing nan to 0 in reviews_per_month col
airbnb[['reviews_per_month']] = airbnb[['reviews_per_month']].fillna(0)

In [28]:
airbnb.isnull().sum()

id                                    0
name                                 16
host_id                               0
host_name                            21
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       10052
reviews_per_month                     0
calculated_host_listings_count        0
availability_365                      0
dtype: int64

In [29]:
import datetime

In [31]:
airbnb.last_review.sample(15)

44012   2019-12-05
21446          NaT
29682   2019-06-17
7019           NaT
7079    2019-06-22
31604   2019-01-07
3904    2018-04-24
25890   2018-10-31
1868    2019-06-19
48504          NaT
14591   2019-06-15
9638    2015-02-11
26169   2019-06-30
5918    2018-06-30
29436   2019-06-14
Name: last_review, dtype: datetime64[ns]

In [32]:
# changing NaT to a dummy date - 1970/1/1
airbnb.last_review = airbnb.last_review.fillna(datetime.datetime(1970,1,1))

In [33]:
airbnb.last_review.describe()

<ipython-input-33-7dfdc1555c78>:1: FutureWarning: Treating datetime data as categorical rather than numeric in `.describe` is deprecated and will be removed in a future version of pandas. Specify `datetime_is_numeric=True` to silence this warning and adopt the future behavior now.
  airbnb.last_review.describe()


count                   48895
unique                   1765
top       1970-01-01 00:00:00
freq                    10052
first     1970-01-01 00:00:00
last      2019-12-06 00:00:00
Name: last_review, dtype: object

In [44]:
# we have handeled all the data
airbnb.isnull().sum()

id                                0
name                              0
host_id                           0
host_name                         0
neighbourhood_group               0
neighbourhood                     0
latitude                          0
longitude                         0
room_type                         0
price                             0
minimum_nights                    0
number_of_reviews                 0
last_review                       0
reviews_per_month                 0
calculated_host_listings_count    0
availability_365                  0
dtype: int64

<h1><b>Analysis</b></h1>

In [43]:
airbnb.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 48858 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype         
---  ------                          --------------  -----         
 0   id                              48858 non-null  int64         
 1   name                            48858 non-null  object        
 2   host_id                         48858 non-null  int64         
 3   host_name                       48858 non-null  object        
 4   neighbourhood_group             48858 non-null  object        
 5   neighbourhood                   48858 non-null  object        
 6   latitude                        48858 non-null  float64       
 7   longitude                       48858 non-null  float64       
 8   room_type                       48858 non-null  object        
 9   price                           48858 non-null  int64         
 10  minimum_nights                  48858 non-null  int64         
 11  nu